rerun_cpp/src/rerun/status.cpp

```diff
@@ -9,6 +10,66 @@ namespace rerun {
     Status::Status(const rr_status& status)
         : code(static_cast<StatusCode>(status.code)), description(status.description) {}
 
+    Status::Status(const arrow::Status& status) {
+        switch (status.code()) {
+            case arrow::StatusCode::OK:
+                code = StatusCode::Ok;
+                break;
+            case arrow::StatusCode::OutOfMemory:
+                code = StatusCode::OutOfMemory;
+                break;
+            case arrow::StatusCode::KeyError:
+                code = StatusCode::ArrowStatusCode_KeyError;
+                break;
+            case arrow::StatusCode::TypeError:
+                code = StatusCode::ArrowStatusCode_TypeError;
+                break;
+            case arrow::StatusCode::Invalid:
+                code = StatusCode::ArrowStatusCode_Invalid;
+                break;
+            case arrow::StatusCode::IOError:
+                code = StatusCode::ArrowStatusCode_IOError;
+                break;
+            case arrow::StatusCode::CapacityError:
+                code = StatusCode::ArrowStatusCode_CapacityError;
+                break;
+            case arrow::StatusCode::IndexError:
+                code = StatusCode::ArrowStatusCode_IndexError;
+                break;
+            case arrow::StatusCode::Cancelled:
+                code = StatusCode::ArrowStatusCode_Cancelled;
+                break;
+            case arrow::StatusCode::UnknownError:
+                code = StatusCode::ArrowStatusCode_UnknownError;
+                break;
+            case arrow::StatusCode::NotImplemented:
+                code = StatusCode::ArrowStatusCode_NotImplemented;
+                break;
+            case arrow::StatusCode::SerializationError:
+                code = StatusCode::ArrowStatusCode_SerializationError;
+                break;
+            case arrow::StatusCode::RError:
+                code = StatusCode::ArrowStatusCode_RError;
+                break;
+            case arrow::StatusCode::CodeGenError:
+                code = StatusCode::ArrowStatusCode_CodeGenError;
+                break;
+            case arrow::StatusCode::ExpressionValidationError:
+                code = StatusCode::ArrowStatusCode_ExpressionValidationError;
+                break;
+            case arrow::StatusCode::ExecutionError:
+                code = StatusCode::ArrowStatusCode_ExecutionError;
+                break;
+            case arrow::StatusCode::AlreadyExists:
+                code = StatusCode::ArrowStatusCode_AlreadyExists;
+                break;
+            default:
+                code = StatusCode::Unknown;
```

Comment:
```suggestion
                code = StatusCode:: ArrowStatusCode_Unknown;
```

Reply:
I'd argue that's different - we don't know if it's an error unknown, instead it is an error that we literally don't know about